src/find/matchers/ls.rs

```diff
@@ -0,0 +1,256 @@
+// This file is part of the uutils findutils package.
+//
+// For the full copyright and license information, please view the LICENSE
+// file that was distributed with this source code.
+
+use chrono::DateTime;
+use std::{
+    fs::File,
+    io::{stderr, Write},
+};
+use walkdir::DirEntry;
+
+use super::{Matcher, MatcherIO};
+
+#[cfg(unix)]
+fn format_permissions(mode: u32) -> String {
+    let file_type = if mode & 0o170000 == 0o040000 {
```

Comment:
It would be better to use symbolic constants like `mode & libc::S_IFMT == libc::S_IFDIR`

Reply:
thanks. fixed them in the new commit.